apps/myaccount/src/components/profile/profile.tsx

```diff
@@ -685,12 +857,299 @@ export const Profile: FunctionComponent<ProfileProps> = (props: ProfileProps): R
+
+            if (existingPrimaryEmail && !existingEmailList.includes(existingPrimaryEmail)) {
+                existingEmailList.push(existingPrimaryEmail);
+                data.Operations.push({
+                    op: "replace",
+                    value: {
+                        [schema.schemaId] : {
+                            [EMAIL_ADDRESSES_ATTRIBUTE]: existingEmailList
+                        }
+                    }
+                });
+            }
+        } else if (schema.name === MOBILE_NUMBERS_ATTRIBUTE) {
+
+            data.Operations[0].value = {
+                [ProfileConstants.SCIM2_SCHEMA_DICTIONARY.get("PHONE_NUMBERS")]: [
+                    {
+                        type: "mobile",
+                        value
+                    }
+                ]
+            };
+
+            const existingPrimaryMobile: string = profileInfo.get(MOBILE_ATTRIBUTE);
+            const existingMobileList: string[] = profileInfo?.get(MOBILE_NUMBERS_ATTRIBUTE)?.split(",") || [];
+
+            if (existingPrimaryMobile && !existingMobileList.includes(existingPrimaryMobile)) {
+                existingMobileList.push(existingPrimaryMobile);
+                data.Operations.push({
+                    op: "replace",
+                    value: {
+                        [schema.schemaId] : {
+                            [MOBILE_NUMBERS_ATTRIBUTE]: existingMobileList
+                        }
+                    }
+                });
+            }
+        }
+        updateProfileInfo(data).then((response: AxiosResponse) => {
+            if (response.status === 200) {
+                onAlertFired({
+                    description: t(
+                        "myAccount:components.profile.notifications.updateProfileInfo.success.description"
+                    ),
+                    level: AlertLevels.SUCCESS,
+                    message: t(
+                        "myAccount:components.profile.notifications.updateProfileInfo.success.message"
+                    )
+                });
+
+                // Re-fetch the profile information
+                dispatch(getProfileInformation(true));
+            }
+        }).catch((error: any) => {
+            onAlertFired({
+                description: error?.detail ?? t(
+                    "myAccount:components.profile.notifications.updateProfileInfo.genericError.description"
+                ),
+                level: AlertLevels.ERROR,
+                message: error?.message ?? t(
+                    "myAccount:components.profile.notifications.updateProfileInfo.genericError.message"
+                )
+            });
+        }).finally(() => {
+            setIsSubmitting(false);
+        });
+    };
+
+    /**
+     * Delete a multi-valued attribute value.
+     *
+     * @param schema - schema of the attribute
+     * @param value - value of the attribute
+     */
+    const handleMultiValuedItemDelete = (schema: ProfileSchema, value: string) => {
```

Comment:
Shall we move this to a util file?

Reply:
This will be tracked through 
- https://github.com/wso2/product-is/issues/21633